sdk/python/foundation-models/system/inference/text-generation/llama-files/score/hf-tgi/score.py

```diff
@@ -0,0 +1,460 @@
+DTYPE = "DTYPE"
+TRUST_REMOTE_CODE = "TRUST_REMOTE_CODE"
+MAX_CONCURRENT_REQUESTS = "MAX_CONCURRENT_REQUESTS"
+MAX_BEST_OF = "MAX_BEST_OF"
+MAX_STOP_SEQUENCES = "MAX_STOP_SEQUENCES"
+MAX_INPUT_LENGTH = "MAX_INPUT_LENGTH"
+MAX_TOTAL_TOKENS = "MAX_TOTAL_TOKENS"
+
+# client init env vars
+CLIENT_TIMEOUT = "TIMEOUT"
+MAX_REQUEST_TIMEOUT = 90  # 90s
+
+
+class SupportedTask:
+    """Supported tasks by text-generation-inference"""
+    TEXT_GENERATION = "text-generation"
+    CHAT_COMPLETION = "chat-completion"
+
+
+# default values
+MLMODEL_PATH = "mlflow_model_folder/MLmodel"
+DEFAULT_MODEL_ID_PATH  = "mlflow_model_folder/data"
+client = None
+task_type = SupportedTask.TEXT_GENERATION
+
+
+def run_command(cmd: str) -> Tuple[int, str]:
+    """Run the command and returns the result."""
+    logger.info(f"run_command: executing {cmd}")
+    result = subprocess_run(
+        cmd,
+        shell=True,
+        stdout=PIPE,
+        stderr=STDOUT,
+        encoding=sys.stdout.encoding,
+        errors="ignore",
+    )
+    return result
+
+
+def is_server_healthy():
+    # use psutil to go through active process 
+    WAIT_TIME = 20
+    RETRY_COUNT = 5
+    count = 0
+    while count < RETRY_COUNT and TEXT_GEN_LAUNCHER_PROCESS_NAME not in [p.name() for p in psutil.process_iter()]:
+        logger.warning(f"Process {TEXT_GEN_LAUNCHER_PROCESS_NAME} is not running. Sleeping for {WAIT_TIME}s and retrying")
+        time.sleep(WAIT_TIME)
+        count += 1
+    if count >= RETRY_COUNT:
+        total_dur = RETRY_COUNT * WAIT_TIME
+        raise Exception(f"Sever process not running after waiting for {total_dur}. Terminating")
+
+    logger.info(f"Server process {TEXT_GEN_LAUNCHER_PROCESS_NAME} running. Hitting endpoint with 5s delay")
+    time.sleep(5)
+
+    payload_dict = {
+        "inputs": "Meaning of life is",
+        "parameters":{"max_new_tokens":2}
+    }
+
+    json_str = json.dumps(payload_dict)
+
+    try:
+        response = requests.post(
+            url=LOCAL_HOST_URI,
+            data=json_str,
+            headers={
+                "Content-Type": "application/json"
+            }
+        )
+        logger.info(f"response status code: {response.status_code}")
+        if response.status_code == 200 or response.status_code == 201:
+            return True
+    except Exception as e:
+        logger.warning(f"Test request failed. Error {e}")
+    return False
+
+
+def init():
+    global client
+    global task_type
+
+    try:
+        model_id = os.environ.get(MODEL_ID, DEFAULT_MODEL_ID_PATH)
+        client_timeout = os.environ.get(CLIENT_TIMEOUT, MAX_REQUEST_TIMEOUT)
+
+        for k, v in os.environ.items():
+            logger.info(f"env: {k} = {v}")
+
+        model_path = os.path.join(os.getenv("AZUREML_MODEL_DIR", ""), model_id)
+        model_path = os.path.join(model_path, "model")  # default model path for MLFlow model
+        # Use safetensors if present
+        if os.path.isdir(os.path.join(model_path, "safetensors")):
```

Comment:
I see in the preview registry the folder structure is mlflow_model_folder/data has model and safetensors folder. If that is the final model structure, should this line 133 be appending from line 130 instead of 131?

Reply:
Ooh right. Will need to fix that. @novaturient95  fyi.
Btw, we're testing again with a single folder structure (all files in the same model dir), that's easier to maintain plus follows HF models in open source.

Reply:
Addressed